lib/gui/app/components/target-selector/target-selector-modal.tsx

```diff
@@ -0,0 +1,409 @@
+import { Modal } from '../../styled-components';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+export interface DrivelistTarget extends DrivelistDrive {
+	displayName: string;
+	progress: number;
+	device: string;
+	link: string;
+	linkTitle: string;
+	linkMessage: string;
+	linkCTA: string;
+}
+
+/**
+ * @summary Get a drive's compatibility status object(s)
+ *
+ * @description
+ * Given a drive, return its compatibility status with the selected image,
+ * containing the status type (ERROR, WARNING), and accompanying
+ * status message.
+ */
+function getDriveStatuses(
+	drive: DrivelistTarget,
+): Array<{ type: number; message: string }> {
+	return getDriveImageCompatibilityStatuses(drive, getImage());
+}
+
+const TargetsTable = styled(({ refFn, ...props }) => {
+	return <BaseTable<DrivelistTarget> ref={refFn} {...props}></BaseTable>;
+})`
+	[data-display='table-head']
+		[data-display='table-row']
+		> [data-display='table-cell']:first-child {
+		padding-left: 15px;
+	}
+
+	[data-display='table-head']
+		[data-display='table-row']
+		> [data-display='table-cell'] {
+		padding: 6px 8px;
+		color: #2a506f;
+	}
+
+	[data-display='table-body']
+		> [data-display='table-row']
+		> [data-display='table-cell']:first-child {
+		padding-left: 15px;
+	}
+
+	[data-display='table-body']
+		> [data-display='table-row']
+		> [data-display='table-cell'] {
+		padding: 6px 8px;
+		color: #2a506f;
+	}
+`;
+
+interface DriverlessDrive {
+	link: string;
+	linkTitle: string;
+	linkMessage: string;
+}
+
+interface TargetStatus {
+	message: string;
+	type: number;
+}
+
+function badgeShadeFromStatus(status: string) {
+	switch (status) {
+		case compatibility.containsImage():
+			return 16;
+		case compatibility.system():
+			return 5;
+		default:
+			return 14;
+	}
+}
+
+function renderStatuses(statuses: TargetStatus[]) {
+	return _.map(statuses, (status) => {
+		const badgeShade = badgeShadeFromStatus(status.message);
+		return (
+			<Badge key={status.message} shade={badgeShade}>
+				{status.message}
+			</Badge>
+		);
+	});
+}
+
+const InitProgress = styled(
+	({
+		value,
+		...props
+	}: {
+		value: number;
+		props?: React.ProgressHTMLAttributes<Element>;
+	}) => {
+		return <progress max="100" value={value} {...props}></progress>;
+	},
+)`
+	/* Reset the default appearance */
+	-webkit-appearance: none;
+	appearance: none;
+
+	::-webkit-progress-bar {
+		width: 130px;
+		height: 4px;
+		background-color: #dde1f0;
+		border-radius: 14px;
+	}
+
+	::-webkit-progress-value {
+		background-color: #1496e1;
+		border-radius: 14px;
+	}
+`;
+
+function renderProgress(progress: number) {
+	if (Boolean(progress)) {
+		return (
+			<Flex flexDirection="column">
+				<Txt fontSize={12}>Initializing device</Txt>
+				<InitProgress value={progress} />
+			</Flex>
+		);
+	}
+	return;
+}
+
+interface TableData extends DrivelistTarget {
+	disabled: boolean;
+}
+
+export const TargetSelectorModal = ({
```

Comment:
This could be a `React.Component`

Reply:
Why would it though? With a plain function + useState + useEffect it's just simpler and more readable, I think the issue is the way we handle the current application state *everywhere* and the application state itself needs some refactoring, there are times where I see components called multiple times even though the application state is the same (or should be, but it triggers multiple times)

Reply:
I don't mind. But a 230 lines function is not exactly simple and readable.

Reply:
Touche, I'm probably just being stubborn because with the current application state it's better to have a class. I'll convert in the next push